pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/util/HelixSetupUtils.java

```diff
@@ -175,4 +185,40 @@ private static HelixManager startHelixControllerInStandadloneMode(String helixCl
     return HelixControllerMain
         .startHelixController(zkUrl, helixClusterName, pinotControllerInstanceId, HelixControllerMain.STANDALONE);
   }
+
+  private static void createLeadControllerResourceIfNeeded(String helixClusterName, HelixAdmin admin,
+      boolean enableBatchMessageMode, boolean enableLeadControllerResource) {
+    List<String> resources = admin.getResourcesInCluster(helixClusterName);
```

Comment:
Instead of getting all resources (which can number in 1000s) can we just get the idealstate of the lead controller resource to check if it is there?

Reply:
That's a good idea! Addressed.